packages/core/src/shared/errors.ts

```diff
@@ -243,11 +243,24 @@ export class ToolkitError extends Error implements ErrorInformation {
     }
 }
 
-export function getErrorMsg(err: Error | undefined): string | undefined {
+/**
+ * Derives an error message from the given error object.
+ * Depending on the Error, the property used to derive the message can vary.
+ *
+ * @param withCause If the error is a ToolkitError, the message of the cause will
+ *                  be appended and delimited by a '::'. Eg: msg1::causeMsg1:causeMsg2
+ */
+export function getErrorMsg(err: Error | undefined, withCause = false): string | undefined {
     if (err === undefined) {
         return undefined
     }
 
+    if (withCause && err instanceof ToolkitError) {
```

Comment:
javascript errors can have `cause`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

I think our ES type definitions are just out of date (may change after we update our min vscode).

```suggestion
    if (withCause && (err as any).code) {
```